Essentials/Commands/BlocksModule.cs

```diff
@@ -140,122 +140,175 @@ public void OffSubtype(string subtype)
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_Assembler) ||
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_OxygenGenerator)));
-                    var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
-                    foreach (var item in list)
-                    {
-                        item.Enabled = true;
-                    }
-                    count++;
-
-                }
-                if (general.Contains("wea"))
-                {
-                    status = "Weapon";
-                    var blocks = new List<IMySlimBlock>();
-                    grid.GetBlocks(blocks, f => f.FatBlock != null && f.FatBlock is IMyFunctionalBlock
+                                var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
+                                foreach (var item in list)
+                                {
+                                    item.Enabled = true;
+                                }
+                                count++;
+                            }
+                        }
+                        break;
+                    case blockcategory.weapons:
+                        {
+                            foreach (var entity in MyEntities.GetEntities().OfType<MyCubeGrid>())
+                            {
+                                IMyCubeGrid grid = entity as MyCubeGrid;
+                                var blocks = new List<IMySlimBlock>();
+                                grid.GetBlocks(blocks, f => f.FatBlock != null && f.FatBlock is IMyFunctionalBlock
                     && (f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_InteriorTurret) ||
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_TurretBase) ||
                     f.FatBlock.BlockDefinition.TypeId == typeof(MyObjectBuilder_LargeMissileTurret)));
-                    var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
-                    foreach (var item in list)
-                    {
-                        item.Enabled = true;
-                    }
-                    count++;
+                                var list = blocks.Select(f => (IMyFunctionalBlock)f.FatBlock).Where(f => !f.Enabled).ToArray();
+                                foreach (var item in list)
+                                {
+                                    item.Enabled = true;
+                                }
+                                count++;
+                            }
+                        }
+                        break;
 
                 }
             }
+            else
+            {
+                Context.Respond($"{category} is not part of the set. Use the following with this command: power, production, weapons ");
```

Comment:
Try `string.Join(", ", Enum.Values(typeof(blockcategory));`

Reply:
dang it. I forgot to use it on for that. on it